src/commands/synthetics/__tests__/run-test.test.ts

```diff
@@ -7,6 +7,25 @@ import * as runTests from '../run-test'
 import * as utils from '../utils'
 import {config, mockReporter} from './fixtures'
 
+expect.extend({
+  toBeError(received, errorClass, errorCode) {
+    const pass = received instanceof errorClass && errorCode === received.code
+    if (pass) {
+      return {
+        message: () =>
+          `expected ${received.constructor.name} with code ${received.code} but received ${errorClass.name} with code ${errorCode}`,
+        pass: true,
+      }
+    } else {
+      return {
+        message: () =>
+          `Received ${received.constructor.name} with code ${received.code} but expected ${errorClass.name} Error with code ${errorCode}`,
+        pass: false,
+      }
+    }
+  },
+})
```

Comment:
We could use this package: https://github.com/DanielHreben/jest-matcher-specific-error
Otherwise, let's move this to another file.
In worker, the matchers are in https://github.com/DataDog/synthetics-worker/blob/prod/worker/test/helpers/jest.ts which is then imported by jest as a setup: https://github.com/DataDog/synthetics-worker/blob/prod/jest.config.js#L23

Reply:
I've resorted in using the` jest-matcher-specific-error` package.